src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneIndexingService.cs

```diff
@@ -50,22 +53,46 @@ public class LuceneIndexingService
 
         public ILogger Logger { get; }
 
-        public async Task ProcessContentItemsAsync()
+        public async Task ProcessContentItemsAsync(string indexName = default)
         {
             // TODO: Lock over the filesystem in case two instances get a command to rebuild the index concurrently.
+            var allIndicesStatus = new Dictionary<string, int>();
+            var lastTaskId = Int32.MaxValue;
+            IEnumerable<LuceneIndexSettings> indexSettingsList = null;
 
-            var allIndices = new Dictionary<string, int>();
 
-            // Find the lowest task id to process
-            var lastTaskId = int.MaxValue;
-            foreach (var indexName in _indexManager.List())
+            if (String.IsNullOrEmpty(indexName))
             {
+                indexSettingsList = _luceneIndexSettingsService.List();
```

Comment:
This can cause issue if someone is removing or adding an index while we are indexing.

Reply:
You could use an immutable array in `LuceneIndexSettingsService` without any locking unless for writing the file.

Reply:
Do it if you have time 😉 any help is appreciated at this point I'm way over my planning time on this 😄

Reply:
Okay as i have time, remind me if i forget it ;)